File: app/src/main/java/com/github/kongpf8848/animation/evaluator/HeightEvaluator.java
/**
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.github.kongpf8848.animation.evaluator;

import android.animation.IntEvaluator;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;

public class HeightEvaluator extends IntEvaluator {

    private View v;
    public HeightEvaluator(View v) {
        this.v = v;
    }

    @NonNull
    @Override
    public Integer evaluate(float fraction, Integer startValue, Integer endValue) {
        int num = super.evaluate(fraction, startValue, endValue);
        ViewGroup.LayoutParams params = v.getLayoutParams();
        params.height = num;
        v.setLayoutParams(params);
        return num;
    }
}

File: app/src/main/java/com/github/kongpf8848/animation/evaluator/HeightEvaluator.kt
package com.github.kongpf8848.animation.evaluator

import android.animation.IntEvaluator
import android.view.View

class HeightEvaluator(private val v: View) : IntEvaluator() {

    override fun evaluate(fraction: Float, startValue: Int, endValue: Int): Int {
        val num = super.evaluate(fraction, startValue, endValue)
        val params = v.layoutParams
        params.height = num
        v.layoutParams = params
        return num
    }

}